crates/common/src/header.rs

```diff
@@ -34,6 +35,14 @@ impl BlockHeader {
             .with_number(self.number + 1)
             .with_parent_hash(self.hash)
     }
+
+    /// Creates a [StateUpdate] with the block hash and state commitment fields intialized
+    /// to match this header.
+    pub fn init_state_update(&self) -> StateUpdate {
+        StateUpdate::default()
+            .with_block_hash(self.hash)
+            .with_state_commitment(self.state_commitment)
+    }
```

Comment:
The name of this method could be better... suggestions? Could also be `StateUpdate::from(&BlockHeader)` but I don't like that much.